aws/efs_test.go

```diff
@@ -0,0 +1,251 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/efs"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/retry"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func findEFSVolumesByNameTag(t *testing.T, session *session.Session, name string) []*string {
+	output, err := efs.New(session).DescribeFileSystems(&efs.DescribeFileSystemsInput{})
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	var volumeIds []*string
+	for _, volume := range output.FileSystems {
+		if aws.StringValue(volume.Name) == name {
+			volumeIds = append(volumeIds, volume.FileSystemId)
+		}
+	}
+
+	return volumeIds
+}
+
+func createTestEFSVolumeMountTarget(t *testing.T, session *session.Session, fileSystemId *string, subnetId *string) efs.MountTargetDescription {
+	svc := efs.New(session)
+
+	mountTarget, err := svc.CreateMountTarget(&efs.CreateMountTargetInput{
+		FileSystemId: fileSystemId,
+		SubnetId:     subnetId,
+	})
+
+	if err != nil {
+		assert.Failf(t, "Could not create test EFS mount target for volume", errors.WithStackTrace(err).Error())
+	}
+
+	return *mountTarget
+}
+
+func createTestEFSVolume(t *testing.T, session *session.Session, name string, az string) efs.FileSystemDescription {
+	svc := efs.New(session)
+
+	volume, err := svc.CreateFileSystem(&efs.CreateFileSystemInput{
+		AvailabilityZoneName: aws.String(az),
+		Tags: []*efs.Tag{
+			{
+				Key:   aws.String("Name"),
+				Value: aws.String(name),
+			},
+		},
+	})
+
+	if err != nil {
+		assert.Failf(t, "Could not create test EFS volume", errors.WithStackTrace(err).Error())
+	}
+
+	if err := retry.DoWithRetry(
+		logging.Logger,
+		fmt.Sprintf("Waiting until EFS volume ID %s is fully created", *volume.FileSystemId),
+		10,
+		1*time.Second,
+		func() error {
+			details, err := svc.DescribeFileSystems(&efs.DescribeFileSystemsInput{
+				FileSystemId: volume.FileSystemId,
+			})
+			if err != nil {
+				return err
+			}
+			if len(details.FileSystems) <= 0 || len(details.FileSystems) >= 2 {
```

Comment:
Not sure if make sense to check length of array field [`FileSystems`](https://docs.aws.amazon.com/sdk-for-go/api/service/efs/#DescribeFileSystemsOutput) `<=0` since it can't go below 0

Maybe make sense to check for `!=1`?